ARTeeView/ViewController.m

```diff
@@ -57,9 +55,144 @@ - (void)didReceiveMemoryWarning {
+    
+    if (bestMser)
+    {
+        NSLog(@"minDist: %f", bestPoint);
+        
+        cv::Rect bound = cv::boundingRect(*bestMser);
+        cv::rectangle(image, bound, 201, 3);
+    }
+    else 
+    {
+        cv::rectangle(image, cv::Rect(0, 0, W, H), 255, 3);
+    }
+
+    
+    return;
     const char* str = [@"Toptal" cStringUsingEncoding: NSUTF8StringEncoding];
-    cv::putText(image, str, cv::Point(100, 100), CV_FONT_HERSHEY_PLAIN, 2.0, cv::Scalar(0, 0, 255));
+    cv::putText(image, str, cv::Point(100, 100), CV_FONT_HERSHEY_PLAIN, 2.0, cv::Scalar(0, 0, 255));\
+    cvCloneImage((IplImage *)(&image)->data);
+    return;
+    const char* cascadeFileFace = "haarcascades\\haarcascade_frontalface_alt.xml";	// Path to the Face Detection HaarCascade XML file
+    CvHaarClassifierCascade *cascadeFace = (CvHaarClassifierCascade *) cvLoad(cascadeFileFace);
+    IplImage *image4 = cvCloneImage((IplImage *)&image);
+    CvMat image2 = image;
+    IplImage image3 = image;
+    CvSize size = cvSize(100, 200);
+    IplImage *imageInHSV = cvCreateImage(size, 8, 3);
+    cvCvtColor(image4, imageInHSV, CV_BGR2HSV);	// (note that OpenCV stores RGB images in B,G,R order.
+    IplImage* imageDisplayHSV = cvCreateImage(cvGetSize(&image), 8, 3);	// Create an empty HSV image
+    //cvSet(imageDisplayHSV, cvScalar(0,0,0, 0));	// Clear HSV image to blue.
+    int hIn = imageDisplayHSV->height;
+    int wIn = imageDisplayHSV->width;
+    int rowSizeIn = imageDisplayHSV->widthStep;		// Size of row in bytes, including extra padding
+    char *imOfsDisp = imageDisplayHSV->imageData;	// Pointer to the start of the image HSV pixels.
+    char *imOfsIn = imageInHSV->imageData;	// Pointer to the start of the input image HSV pixels.
+    for (int y=0; y<hIn; y++) {
+        for (int x=0; x<wIn; x++) {
+            // Get the HSV pixel components
+            uchar H = *(uchar*)(imOfsIn + y*rowSizeIn + x*3 + 0);	// Hue
+            uchar S = *(uchar*)(imOfsIn + y*rowSizeIn + x*3 + 1);	// Saturation
+            uchar V = *(uchar*)(imOfsIn + y*rowSizeIn + x*3 + 2);	// Value (Brightness)
+            // Determine what type of color the HSV pixel is.
+            int ctype = getPixelColorType(H, S, V);
+            //ctype = x / 60;
+            // Show the color type on the displayed image, for debugging.
+            *(uchar*)(imOfsDisp + (y)*rowSizeIn + (x)*3 + 0) = cCTHue[ctype];	// Hue
+            *(uchar*)(imOfsDisp + (y)*rowSizeIn + (x)*3 + 1) = cCTSat[ctype];	// Full Saturation (except for black & white)
+            *(uchar*)(imOfsDisp + (y)*rowSizeIn + (x)*3 + 2) = cCTVal[ctype];		// Full Brightness
+        }
+    }
+    // Display the HSV debugging image
+    IplImage *imageDisplayHSV_RGB = cvCreateImage(cvGetSize(imageDisplayHSV), 8, 3);
+    cvCvtColor(imageDisplayHSV, imageDisplayHSV_RGB, CV_HSV2BGR);	// (note that OpenCV stores RGB images in B,G,R order.
+    cvNamedWindow("Colors", 1);
+    cvShowImage("Colors", imageDisplayHSV_RGB);
+     */
```

Comment:
I think it's useful to remember our failures, so we don't forget the lessons they taught us.